benchmark/benchmark_test.go

```diff
@@ -37,3 +37,21 @@ func BenchmarkPureFib(b *testing.B) {
 	}
 	result = r
 }
+
+func BenchmarkCollatz(b *testing.B) {
```

Comment:
This benchmark sums up the first 100 [collatz numbers](https://en.wikipedia.org/wiki/Collatz_conjecture). 

Results on my computer are as follows:
```
BenchmarkCollatz-4       	    2750	   4459093 ns/op
BenchmarkPureCollatz-4   	   12122	    963438 ns/op
```